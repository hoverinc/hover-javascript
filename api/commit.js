module.exports = require('../dist/api/commit')
