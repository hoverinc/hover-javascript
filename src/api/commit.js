const {readdirSync, statSync} = require('fs')
const {join, sep} = require('path')

//
// Common scopes

/**
 * Build scopes
 * - commit
 * - deps
 * - deps-dev
 * - format
 * - lint
 */
const build = () => ['commit', 'deps', 'deps-dev', 'format', 'lint']

const scopes = {build}

//
// Scope helpers

/**
 * Enumerate config files
 *
 * Example
 *
 * - `dir/foo.js`
 * - `dir/bar.config.js`
 * - `dir/baz.json`
 * - `dir/qux.ts`
 *
 * `ls.config('./dir') â†’ ['dir/foo', 'dir/bar', 'dir/baz', 'dir/qux']`
 *
 * @param {string} path - directory to enumerate
 */
const configs = path =>
  readdirSync(path)
    .filter(f => statSync(join(path, f)).isFile())
    .map(
      f =>
        `${path.split(sep).reverse()[0]}/${f.replace(
          /(\.config)?.(json|js|ts)/,
          '',
        )}`,
    )

const ls = {configs}

module.exports = {ls, scopes}
