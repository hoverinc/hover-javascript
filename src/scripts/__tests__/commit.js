import cases from 'jest-in-case'
import {
  unquoteSerializer,
  winPathSerializer,
  relativePathSerializer,
} from '../../test/helpers/serializers'

jest.mock('commitizen/dist/cli/git-cz')

expect.addSnapshotSerializer(unquoteSerializer)
expect.addSnapshotSerializer(winPathSerializer)
expect.addSnapshotSerializer(relativePathSerializer)

cases(
  'commit',
  ({args = [], env = {}}) => {
    // beforeEach
    const {bootstrap: bootstrapMock} = require('commitizen/dist/cli/git-cz')

    const originalArgv = process.argv
    const originalExit = process.exit
    const originalEnv = process.env

    process.exit = jest.fn()
    process.argv = ['node', '../commit', ...args]
    process.env = env

    try {
      // tests
      require('../commit')

      expect(bootstrapMock).toHaveBeenCalledTimes(1)

      const [call] = bootstrapMock.mock.calls
      const [optionsArg, argsArg] = call

      expect(optionsArg).toMatchSnapshot()
      expect(argsArg.join(' ')).toMatchSnapshot()
    } catch (error) {
      throw error
    } finally {
      // afterEach
      process.exit = originalExit
      process.argv = originalArgv
      process.env = originalEnv

      jest.resetModules()
    }
  },
  {
    'bootstraps @commitlint/prompt': {},
    'forwards arguments': {
      args: ['--retry'],
    },
  },
)
