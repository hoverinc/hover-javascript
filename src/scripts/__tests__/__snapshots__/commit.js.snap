// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commit bootstraps @commitlint/prompt 1`] = `
Object {
  cliPath: <PROJECT_ROOT>/node_modules/commitizen,
  config: Object {
    path: @commitlint/prompt,
  },
}
`;

exports[`commit bootstraps @commitlint/prompt 2`] = `node ../commit`;

exports[`commit forwards arguments 1`] = `
Object {
  cliPath: <PROJECT_ROOT>/node_modules/commitizen,
  config: Object {
    path: @commitlint/prompt,
  },
}
`;

exports[`commit forwards arguments 2`] = `node ../commit --retry`;

exports[`commit strips errant "commit" argument 1`] = `
Object {
  cliPath: <PROJECT_ROOT>/node_modules/commitizen,
  config: Object {
    path: @commitlint/prompt,
  },
}
`;

exports[`commit strips errant "commit" argument 2`] = `node ../commit`;

exports[`commit strips errant "commit" argument and forwards arguments 1`] = `
Object {
  cliPath: <PROJECT_ROOT>/node_modules/commitizen,
  config: Object {
    path: @commitlint/prompt,
  },
}
`;

exports[`commit strips errant "commit" argument and forwards arguments 2`] = `node ../commit --retry`;
